### PR TITLE
Port #47620 to master

### PR DESCRIPTION
**Pull request by Ch3LL**
Port #47620 to master

### CODE REVIEW

**Comment by Ch3LL** [comment]
ping @cbosdo would you mind reviewing here. I had to rebase with the changes you recently got merged and make some changes from the original PR to ensure loader worked with the new defined function as well.

**Comment by cbosdo** [comment]
@Ch3LL I completely forgot about that PR... and there are some bits in the UEFI / loader handling that would need to be rethinked.

* For libvirt before 5.4.0 the user would need to provide both the loader and the nvram template for most UEFI cases (maybe not bhyve) and after, just a simple flag saying UEFI is wanted will suffice. For this I would move the `virt.init()` `loader` parameter in the `boot` one. This would allow adding `nvram` and other boot-related params later.

* This PR only touches the `virt.init()` function, but not the `virt.update()` one... and thus there will be problems when reapplying the `virt.running` or `virt.defined` states.

How do you see this backport PR? Just stick to what was in the original PR and then file another PR to fix the above mentioned points or add the above points to this PR? In the latter case I would probably have to take over your PR to complete and fix it.

File: salt/states/virt.py [hunk 1]
```diff
@@ -520,6 +523,12 @@ def running(
         See :ref:`init-graphics-def` for more details on this dictionary
 
         .. versionadded:: 2019.2.0
+    :param loader:
+        Firmware loader for the new virtual machine.
+        See the **Loader Definition** section of the :py:func:`virt.init
+        <salt.modules.virt.init>` function for more details on this dictionary.
+
+        .. versionadded:: Sodium
```
Comment (cbosdo) [inline salt/states/virt.py:531]: That parameter documentation should also be in the `defined()` state

File: tests/unit/modules/test_virt.py [hunk 2]
```diff
@@ -393,6 +403,56 @@ def test_gen_xml_spice_default(self):
...
+            diskp,
+            nicp,
+            "kvm",
+            "hvm",
+            "x86_64",
+            loader={"readonly": "yes", "type": "pflash", "secure": "yes"},
+        )
+        root = ET.fromstring(xml_data)
+        self.assertFalse("loader" in root.find("os"))
+
```
Comment (cbosdo) [inline tests/unit/modules/test_virt.py:455]: Missing a test case for `virt.update` with loader defined

**Review by cbosdo — CHANGES_REQUESTED** [review]
Apart from the missing documentation in the new `virt.defined` state, the backport is matching the original PR. However as mentioned in the previous comment, the original PR would have needed changes. These may be lifted in a subsequent PR.

**Comment by cbosdo** [comment]
For reference the UEFI part of this PR overlaps with merged PR #56613

**Comment by cbosdo** [comment]
@Ch3LL PR #56882 is my attempt at the backport removing all the loader parameters since those have been added in the `boot` parameter already. The only missing thing is the possibility to set the `secure="yes"` flag. But I think I will have a go at it later

**Comment by cbosdo** [comment]
since the other PR has been merged, this one can be closed

**Comment by Ch3LL** [comment]
Thanks your awesome!

**Pull request closed without merging**